synapse/handlers/federation.py

```diff
@@ -786,13 +801,18 @@ def backfill(self, dest, room_id, limit, extremities):
             })
 
         for e_id in events_to_state:
+            # For paranoia we ensure that these events are marked as
+            # non-outliers
+            ev = event_map[e_id]
+            ev.internal_metadata.outlier = False
```

Comment:
this sort of thing always alarms me slightly; if `outlier` was set doesn't it mean that something else has gone wrong? shouldn't we just raise an AssertionError or something?

Reply:
Assert is actually probably what we want here, yes